cmd/scheduled-task-adder/main.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"cmp"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/google/uuid"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/donor/donordata"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/dynamo"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/identity"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/scheduled"
+)
+
+var taskCount int
+var entryPoint string
+
+type TaskCountEvent struct {
+	TaskCount int `json:"taskCount"`
+}
+
+func init() {
+	flag.IntVar(&taskCount, "taskCount", 0, "Number of scheduled tasks to generate")
+}
+
+func handleAddScheduledTasks(ctx context.Context, taskCountEvent TaskCountEvent) error {
+	flag.Parse()
+
+	if taskCount == 0 {
+		taskCount = taskCountEvent.TaskCount
+	}
+
+	cfg, err := config.LoadDefaultConfig(ctx)
+
+	if err != nil {
+		return fmt.Errorf("failed to load default config: %w", err)
+	}
+
+	awsBaseURL := cmp.Or(os.Getenv("AWS_BASE_URL"), "http://localstack:4566")
+
+	cfg.BaseEndpoint = aws.String(awsBaseURL)
+
+	if awsBaseURL == "http://localhost:4566" {
+		cfg.Credentials = credentials.NewStaticCredentialsProvider(
+			"test",
+			"test",
+			"test",
+		)
+
+		cfg.Region = "eu-west-1"
+	}
+
+	client, err := dynamo.NewClient(cfg, "lpas")
+	if err != nil {
+		return fmt.Errorf("failed to create dynamo client: %w", err)
+	}
+
+	const batchSize = 100
+	donorChan := make(chan any, batchSize)
+	eventChan := make(chan any, batchSize)
+
+	start := time.Now()
+
+	go func() {
```

Comment:
I don't see why you've got this in a go routine with channels? But also what is the plan for this long term being a lambda?

Reply:
It was originally to try to get the time taken to add 1000s of tasks but I've just done a quick comparison with the existing version on main and the run times are within a second of each other 🤦  I'll revert to the original implementation.

I'm planning on having a CI task run to add a large amount of scheduled tasks and then hit a deployed version of the runner lambda so we can keep an eye on performance over time as we add more actions to it. It doesn't necessarily have to run in a lambda but it felt easier than applying CI permissions to run the adder against real AWS.